src/lib/RpcApi.ts

```diff
@@ -276,52 +290,59 @@ export default class RpcApi {
     private _registerKeyguardApis(commands: KeyguardCommand[]) {
         for (const command of commands) {
             // Server listener
-            this._keyguardClient.on(command, (result, state) => {
-                // Recover state
-                this._recoverState(state);
+            this._keyguardClient.on(command,
+                (result, state) => this._keyguardSuccessHandler(command, result, state),
+                (error, state) => this._keyguardErrorHandler(command, error, state),
+            );
+        }
+    }
 
-                // Set result
-                this._store.commit('setKeyguardResult', result);
+    private _keyguardSuccessHandler(command: KeyguardCommand, result: KeyguardResult, state?: ObjectType | null) {
```

Comment:
Alternatively I think this could also work and would make a combined type unnecessary here:

```suggestion
    private _keyguardSuccessHandler<C extends KeyguardCommand>(command: C, result: ResultByCommand<C>, state?: ObjectType | null) {
```

Reply:
This! I love this! :heart: :+1: